app/questionnaire/questionnaire_schema.py

```diff
@@ -616,54 +613,23 @@ def _update_answer_dependencies_for_value_source(
     def _update_answer_dependencies_for_list_source(
         self, *, block_id: str, list_name: str
     ) -> None:
-        """Updates dependencies for a block depending on a list collector
-
-        This method also stores a map of { block_depending_on_list_source -> {add_block, remove_block} }, because:
-        blocks like dynamic_answers, don't directly need to depend on the add_block/remove_block,
-        but a block depending on the dynamic answers might (such as a calculated summary)
-        """
-        list_collectors = self.get_list_collectors_for_list(
-            for_list=list_name,
+        """Store the given block as a dependency of the list"""
```

Comment:
If the functionality is really simple maybe docstring is not needed anymore? Should we used it more for describing our design decisions?

Reply:
Given it is now so simple, and is actually only used in one place, I've removed it entirely and put the code into the single place it is used directly